test/membrane_rtc_engine/hls_endpoint_test.exs

```diff
@@ -0,0 +1,108 @@
+defmodule Membrane.RTC.HLSEndpointTest do
+  use ExUnit.Case
+
+  alias Membrane.RTC.Engine
+  alias Membrane.RTC.Engine.Endpoint.HLS
+  alias Membrane.RTC.Engine.{Message}
+  alias Membrane.RTC.Engine.Support.FileEndpoint
+
+  @fixtures_dir "./test/fixtures/"
+  @reference_dir "./test/hls_reference/"
+  @output_dir "./test/hls_output/"
+
+  setup do
+    options = [
+      id: "test_rtc"
+    ]
+
+    {:ok, pid} = Engine.start_link(options, [])
+
+    Engine.register(pid, self())
+
+    [rtc_engine: pid]
+  end
+
+  describe "HLS Endpoint test" do
+    test "creates correct hls stream from h264 file", %{rtc_engine: rtc_engine} do
+      file_endpoint_id = "file-endpoint-id"
+
+      file_name = "video.h264"
+      file_path = Path.join(@fixtures_dir, file_name)
+
+      hls_endpoint_id = "hls-endpoint"
+
+      track_id = "test-track-id"
+      stream_id = "test-stream"
+
+      track =
+        Engine.Track.new(
+          :video,
+          stream_id,
+          file_endpoint_id,
+          :H264,
+          nil,
+          [:raw],
+          nil,
+          id: track_id
+        )
+
+      hls_endpoint = %HLS{
+        rtc_engine: rtc_engine,
+        owner: self(),
+        output_directory: Path.join(["./", "test", "hls_output"]),
+        target_window_duration: :infinity,
+        framerate: {60, 1}
+      }
+
+      :ok = Engine.add_endpoint(rtc_engine, hls_endpoint, endpoint_id: hls_endpoint_id)
+
+      file_endpoint = %FileEndpoint{
+        rtc_engine: rtc_engine,
+        file_path: file_path,
+        track: track,
+        processing_children: %{
+          parser: %Membrane.H264.FFmpeg.Parser{
+            attach_nalus?: true,
+            skip_until_parameters?: false,
+            framerate: {60, 1}
+          }
+        },
+        processing_link: fn link_builder -> Membrane.ParentSpec.to(link_builder, :parser) end
+      }
+
+      :ok = Engine.add_endpoint(rtc_engine, file_endpoint, endpoint_id: file_endpoint_id)
+
+      assert_receive %Message.MediaEvent{rtc_engine: ^rtc_engine, to: :broadcast, data: data}
+
+      assert %{
+               "type" => "tracksAdded",
+               "data" => %{
+                 "trackIdToMetadata" => %{track_id => nil},
+                 "peerId" => file_endpoint_id
+               }
+             } == Jason.decode!(data)
```

Comment:
This sounds like a bug :thinking: An endpoint that is not correlated with any peer shouldn't publish any ME.
 
Could we write a comment and create an issue to discuss how we should behave?

Reply:
Issue:
https://membraneframework.atlassian.net/browse/MV-102?atlOrigin=eyJpIjoiNjk2N2I2NzI0MjEzNGZkNDgxNTUyMTJlNWNkNmY1MGMiLCJwIjoiaiJ9